### PR TITLE
WIP - Players can only edit own matches (Ref #332)

### PR DESCRIPTION
**Pull request by manner**

### CODE REVIEW

File: app/controllers/matches_controller.rb [hunk 1]
```diff
@@ -48,10 +49,12 @@ def update_points
 
   # GET /matches/1/edit_results
   def edit_results
+    authorize! :edit, @match
   end
 
   # PATCH/PUT /matches/1/update_results
   def update_results
+    authorize! :edit, @match
```
Comment (nikriek) [inline app/controllers/matches_controller.rb]: You can use authorize as a global controller function

**Review by nikriek — CHANGES_REQUESTED** [review]

File: app/controllers/matches_controller.rb [hunk 2]
```diff
@@ -1,5 +1,6 @@
 class MatchesController < ApplicationController
   before_action :set_match, only: [:show, :edit, :update, :update_points, :edit_results, :update_results, :destroy, :add_game_result, :remove_game_result]
+  authorize_resource only: [:edit]
```
Comment (nikriek) [inline app/controllers/matches_controller.rb:3]: Why only edit?

Comment (manner) [reply app/controllers/matches_controller.rb:3]: US 332 only specifies to restrict editing

Comment (nikriek) [reply app/controllers/matches_controller.rb:3]: then create a ticket? maybe ask @PoJo93

File: spec/controllers/matches_controller_spec.rb [hunk 3]
```diff
@@ -33,13 +53,53 @@
...
+    it "shouldn't allow other users to edit the results of a match" do
+      sign_out @user
+      sign_in @other_user
+      match = Match.create! valid_attributes
+      get :edit_results, params: { id: match.to_param }
+      expect(response).not_to be_success
+    end
+
+    it "should allow admin to edit the results of a match" do
+      sign_out @user
```
Comment (nikriek) [inline spec/controllers/matches_controller_spec.rb]: Do you really need to sign out the other user?

**Review by nikriek — APPROVED** [review]

**Pull request merged**